tests/integration/test_self_hosted_runner.py

```diff
@@ -92,3 +97,86 @@ async def test_dispatch_workflow_with_dockerhub_mirror(
                 "A private docker registry is setup as a dockerhub mirror for this self-hosted"
                 " runner."
             ) in logs
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_wait_on_busy_runner_repo_check(
```

Comment:
Isn't this a general test to flush out busy runners? This test would also make sense without the pre job check, right?
Otherwise, please add some more description in the test to make it more explicit for readers of the code.

Reply:
I was looking into a way to test the wait on pre-job. But reading the logs from GitHub does not work. As the logs retrieved by GitHub API are the logs upload to after the job completes, the flushed runner will not be doing that step.
So I have update the to reflect this test is for testing flushing busy runners.